src/utils.test.tsx

```diff
@@ -379,4 +394,123 @@ export async function typeNewNode(
   await screen.findByText(text);
 }
 
+type NodeDescription = [string, (NodeDescription[] | string[])?];
+
+function createNodesAndRelations(
+  plan: Plan,
+  currentRelationsID: LongID | undefined,
+  nodes: NodeDescription[]
+): Plan {
+  return List(nodes).reduce((rdx: Plan, nodeDescription: NodeDescription) => {
+    const currentRelations = currentRelationsID
+      ? getRelationsNoSocial(
+          rdx.knowledgeDBs,
+          currentRelationsID,
+          rdx.user.publicKey
+        )
+      : undefined;
+    const text =
+      typeof nodeDescription === "string"
+        ? nodeDescription
+        : nodeDescription[0];
+    const children =
+      typeof nodeDescription === "string"
+        ? undefined
+        : (nodeDescription[1] as NodeDescription[] | undefined);
+
+    const node = newNode(text, rdx.user.publicKey);
+    const planWithNode = planUpsertNode(rdx, node);
+    // Add Node to current relation
+    const planWithUpdatedRelation = currentRelations
+      ? planUpsertRelations(
+          planWithNode,
+          addRelationToRelations(currentRelations, node.id)
+        )
+      : planWithNode;
+    if (children) {
+      // Create Relations for children
+      const relationForChildren = newRelations(node.id, "", rdx.user.publicKey);
+      const planWithRelations = planUpsertRelations(
+        planWithUpdatedRelation,
+        relationForChildren
+      );
+      return createNodesAndRelations(
+        planWithRelations,
+        relationForChildren.id,
+        children
+      );
+    }
+    return planWithUpdatedRelation;
+  }, plan);
+}
+
+type Options = {
+  activeWorkspace: string;
+};
+
+export function findNodeByText(plan: Plan, text: string): KnowNode | undefined {
+  const { knowledgeDBs, user } = plan;
+  return knowledgeDBs
+    .get(user.publicKey, newDB())
+    .nodes.find((node) => node.text === text);
+}
+
+export async function setupTestDB(
+  appState: TestAppState,
+  nodes: NodeDescription[],
+  options?: Options
+): Promise<Plan> {
+  const plan = createNodesAndRelations(createPlan(appState), undefined, nodes);
+
+  const setNewWorkspace =
+    options?.activeWorkspace && findNodeByText(plan, options.activeWorkspace);
+  const planWithWorkspace = setNewWorkspace
+    ? planUpdateWorkspaces(
+        plan,
+        plan.knowledgeDBs
+          .get(plan.user.publicKey, newDB())
+          .workspaces.push(setNewWorkspace.id),
+        setNewWorkspace.id
+      )
+    : plan;
+  await execute({ ...appState, plan: planWithWorkspace });
+  return planWithWorkspace;
+}
+
+export function startDragging(
+  container: Container,
+  draggableID: string
+): Element {
+  const el = container.querySelector(
+    `div[data-rfd-draggable-id="${draggableID}"]`
+  );
+  if (!el) {
+    throw new Error(`Element with drag id ${draggableID} not found`);
+  }
+  // User event doesn't work to activate dragging on a specific event, the only way to use
```

Comment:
have you tried fireEvent?
`const draggableElement = getByTestId('draggable');
  const droppableElement = getByTestId('droppable');

  fireEvent.dragStart(draggableElement);
  fireEvent.dragOver(droppableElement);
  fireEvent.drop(droppableElement);`

Reply:
I don't see how that would be an improvemnt. `getByTestId` is not a very good way to do things and you still will end up using fireEvent instead of useEvent